### PR TITLE
Update infrastructure & metadata

### PR DESCRIPTION
**Pull request by abitofevrything**
# Description

Update the examples and README for 6.0.0 and clean up metadata/configuration files.

For @l7ssha:
We need to add two new secrets to the repository, in addition to the current `TEST_TOKEN`, as these values are no longer hard coded:
- `TEST_TEXT_CHANNEL` is the ID of the text channel to use for testing (#ci on the Discord server, 846139169818017812)
- `TEST_GUILD` is the ID of the guild to use for testing (the official Discord server, 846136758470443069)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,86 +1,68 @@
...
-#### [Discord API docs](https://discordapp.com/developers/docs/intro)
-Discord API documentation features rich descriptions about all topics that nyxx covers.
+### [Discord API docs](https://discord.dev/)
+Discord's API documentation details what nyxx implements & provides more detailed explanations of certain topics.
 
-#### [Discord API Guild](https://discord.gg/discord-api)
+### [Discord API Server](https://discord.gg/discord-api)
 The unofficial guild for Discord Bot developers. To get help with nyxx check `#dart_nyxx` channel.
 
 #### [Dartdocs](https://www.dartdocs.org/documentation/nyxx/latest/)
```
Comment (Rapougnac) [inline README.md:59]:
```suggestion
#### [Pub.dev docs](https://pub.dev/documentation/nyxx)
```

File: README.md [hunk 2]
```diff
@@ -1,86 +1,68 @@
...
+- [nyxx_extensions](https://pub.dev/packages/nyxx_extensions): Miscellaneous helpers for common situations when developing bots.
 
 ## More examples
 
-Nyxx examples can be found [here](https://github.com/nyxx-discord/nyxx/tree/dev/example).
-
-### Example bots
-- [Running on Dart](https://github.com/l7ssha/running_on_dart)
+- More examples can be found in our GitHub repository [here](https://github.com/nyxx-discord/nyxx/tree/main/example).
+- [Running on Dart](https://github.com/nyxx-discord/running_on_dart) is a complete example of a bot written in nyxx.
```
Comment (Rapougnac) [inline README.md:41]:
```suggestion
- [Running on Dart](https://github.com/nyxx-discord/running_on_dart) is a complete example of a bot written with nyxx.
```
Small nit here, but I think it makes more sense

File: example/example.dart
Comment (Rapougnac) [inline example/example.dart:1]: We shouldn't remove the example file, correct me if I'm wrong, as it's rendered on pub on the example tab.

File: README.md (hunk 1 above)
Comment (Rapougnac) [reply README.md]: @abitofevrything If it's not intentional, can you also remove the last heading mark? Missed that

**Review by l7ssha — APPROVED** [review]

**Pull request merged**